### PR TITLE
Enable syncing of Host LEDs on Split Common

### PR DESCRIPTION
**Pull request by drashna**
## Description

This enables syncing of the host LEDs (caps/num/etc) between halves of split keyboard. 
This allows for things like OLED displays to pull the status. 

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Note: I'm not exactly happy about how this code change looks, and how it handles the host led stuff.  However, request for this sort of functionality comes up from time to time, so it is worth opening, even as is.

**Comment by fauxpark** [comment]
IMO all references to "host LEDs" should be renamed to "indicators", as that seems a little clearer as to its function. But that may be something for another time.

**Comment by drashna** [comment]
Do you mean the defines, the functions, or both?

**Comment by fauxpark** [comment]
Everything. The difference between each of the lighting systems, and the indicator LEDs (basically anything that references LEDs at all), should be clearer and more consistent, however this is a much larger topic I think. For example I'd also like to rename RGBLight to Underglow, and change the keycode prefixes to `UG_`, `LM_`, and `RM_`.

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

File: docs/feature_split_keyboard.md [hunk 1]
```diff
@@ -197,6 +197,12 @@ displaying status on an OLED screen).
 
 This mirrors the master side matrix to the slave side for features that react or require knowledge of master side key presses on the slave side.  This adds a few bytes of data to the split communication protocol and may impact the matrix scan speed when enabled. The purpose of this feature is to support cosmetic use of key events (e.g. RGB reacting to Keypresses).
 
+```c
+#define SPLIT_HOST_SYNC_ENABLE
+```
+
+This enables syncingo of the Host LED status (caps lock, num lock, etc) betwen both halves of the split keyboard. The main purpose of this feature is to enable support for use of things like OLED display of the Host LED status.
```
Comment (BalzGuenat) [inline docs/feature_split_keyboard.md:204]:
```suggestion
This enables syncing of the Host LED status (caps lock, num lock, etc) betwen both halves of the split keyboard. The main purpose of this feature is to enable support for use of things like OLED display of the Host LED status. 
```

**Review by noroadsleft — APPROVED** [review]
`__attribute__ ((weak))` :heavy_check_mark:

100% agree with fauxpark that the LED subsystems should be renamed. Every time I look at source and see references to "host LEDs" I have to think for a moment about what that means. Saying something like "USB LED Indicators" is far more clear.

**Comment by drashna** [comment]
I'm closing this in favor of #11930, as that includes the same implementation.

**Pull request closed without merging**